pkg/datapath/sockets/sockets.go

```diff
@@ -126,6 +137,107 @@ func Destroy(logger *slog.Logger, filter SocketFilter) error {
 	return nil
 }
 
+type BPFSocketDestroyer struct {
```

Comment:
Why split the API into NetlinkSocketDestroyer and BPFSocketDestroyer? This makes things unnecessarily complicated for the caller. `bpf_sock_destroy` being available or not is an implementation detail, not something that needs to leak out of the abstraction.

Reply:
I'm not sure I understand your perspective here. The abstraction is the `SocketDestroyer` interface. All callers just have a `SocketDestroyer`. They don't care which one it is, one is just `Provide`ed at startup based on system capabilities. Sure, you could just have a single `SocketDestroyer` implementation and put some conditional logic inside there to determine which strategy to use, but this feels a bit messier to test and read to me.

Reply:
Makes sense, but let's unexport the implementations then.